### ISSUE TITLE
Modernize responsive object

### ISSUE THREAD

**Issue by thany** [smashingboxes/OwlCarousel2#940]
The `responsive` object appears to be as simple as can be, just a list of pixel widths and corresponding options. First of all, using pixels is outdated. Most importantly, why not allow a full media query?

Secondly, the whole responsiveness mechanism seems to be built around constantly polling the window width, which is slow and limiting. I would *strongly* suggest using `matchMedia`, which works on every mediaquery-supporting browser these days. Although perhaps the current mechanism can still be used for those who really want faux mediaqueries on old browsers.

So in short, I was hoping for something like this:
```
{
  '(max-width: 60em)': {
    //options
  },
  '(max-width: 40em)': {
    //options
  }
}
```
And then, of course, it using `matchMedia` with event listeners and everything so it doesn't clog up the (mobile) browser.
